src/ast/modern/error_definition_processor.ts

```diff
@@ -0,0 +1,31 @@
+import { ASTReader, ASTReaderConfiguration } from "../ast_reader";
+import { ErrorDefinition } from "../implementation/declaration/error_definition";
+import { ParameterList } from "../implementation/meta/parameter_list";
+import { StructuredDocumentation } from "../implementation/meta/structured_documentation";
+import { ModernNodeProcessor } from "./node_processor";
+
+export class ModernErrorDefinitionProcessor extends ModernNodeProcessor<ErrorDefinition> {
+    process(
+        reader: ASTReader,
+        config: ASTReaderConfiguration,
+        raw: any
+    ): ConstructorParameters<typeof ErrorDefinition> {
+        const [id, src, type] = super.process(reader, config, raw);
+
+        const name: string = raw.name;
+        const nameLocation: string | undefined = raw.nameLocation;
+
+        let documentation: string | StructuredDocumentation | undefined;
+
+        if (raw.documentation) {
+            documentation =
+                typeof raw.documentation === "string"
```

Comment:
Given that this should happen in only 0.8.4 and newer, shouldn't this always be a StructuredDocumentation?

Reply:
See previous [related comment](https://github.com/ConsenSys/solc-typed-ast/pull/33#discussion_r623759395).